Overhaul security section (#3022)
jenkins-infra/jenkins.io: content/security/index.adoc

  - link:https://groups.google.com/forum/#!forum/jenkinsci-advisories[`jenkinsci-advisories` list archive]
  - link:https://feeds.feedburner.com/jenkins-security-advisories[`jenkinsci-advisories` RSS feed]

* We send an email notification to the link:https://oss-security.openwall.org/wiki/mailing-lists/oss-security[`oss-security`] mailing list with excerpts of the security advisory.

Additionally, Jenkins administrators are informed about published security issues directly in Jenkins if they have affected versions of Jenkins or plugins installed.

Finally, the Jenkins project is a link:/security/cna[CVE Numbers Authority], and we submit CVE metadata simultaneously with the publication of security advisories, allowing automated security tools using CVE information to identify vulnerable installations.

[IMPORTANT]
====
Even if you run Jenkins on a private network and trust everyone in your team, security issues in Jenkins can still impact you:

- https://en.wikipedia.org/wiki/Cross-site_request_forgery[CSRF vulnerabilities] are a risk even if attackers have no direct access to Jenkins.

> - Does Jenkins build source code you haven't audited, using build scripts someone else wrote, displaying generated reports on its web UI? All of this is a potential a security concern.

Reviewer: ```suggestion
- Does Jenkins build source code you haven't audited, using build scripts someone else wrote, displaying generated reports on its web UI? All of these are potential security concerns.
```

====

[[reporting-vulnerabilities]]
== How to Report a Security Vulnerability

If you find a vulnerability in Jenkins, please report it in the issue tracker under the link:https://issues.jenkins-ci.org/browse/SECURITY[SECURITY project].
This project is configured in such a way that only the reporter, the maintainers, and the Jenkins security team can see the details.
Restricting access to this potentially sensitive information allows core and plugin maintainers to develop effective security fixes that are safe to apply.
We provide issue reporting guidelines and an overview of our process on link:reporting[Reporting Security Vulnerabilities].

If you are unable to report using our issue tracker, you can also send your report to the private Jenkins Security Team mailing list:
`jenkinsci-cert@googlegroups.com`

> To show our appreciation for your help, we'll send you https://wiki.jenkins.io/display/JENKINS/Rewards+for+reporting+security+issues[a small reward] for privately reported, valid vulnerability reports.

Reviewer: Would be nice in the future to bring that page into the jenkins.io site as well.  Separate effort, not required before merging this page.

Reviewer 2: Daniel was working on this, but decided to bound the effort and get what he had complete moving forward. I'm sure he'll get back to some of those other bits, including this one sometime.

Author: Yep, it's coming once this is merged. There's also another link to the wiki somewhere whose label weirdly now is the upcoming page title on jenkins.io 😅

> == Learn More

Reviewer: I absolutely love the formatting of this section.  The layout is easy to read, the short descriptions of the terms are good for many readers, and the hyperlinks lead to much more detailed materials.  Thanks very much for such an elegant use of a definition list for navigation and concepts.
